### ISSUE TITLE
No Data displayed

### ISSUE THREAD

**Issue by madcoda9000** [ekondur/ef-datatable.js#90]
**Describe the bug**
I've setup up everything as in the get Started guide:

No error is thrown (not in browser console and not in vscode)

The grid is shown on the page but contains no data even i can see that the json data is passed to the view from the controller

**To Reproduce**

Controller:

```
[HttpPost]
        public JsonResult GetUsers(DataRequest request) {
            DataResult<AppUser> result = _userManager.Users.ToDataResult(request);
            return Json(result);
        }
```
        
View:

```
@(Html.JS().Datatable<AppUser>()
            .Name("PersonGrid")
            .Columns(cols =>
            {
                cols.Field(a=>a.Id).Visible(true);
            })
            .URL(Url.Action("GetUsers"), "POST")
            .ServerSide(true)
            .Render()
        )
```

**Expected behavior**
as there is no error thrown i would expect that the data is displayed.

**Screenshots**
<img width="2388" alt="Bildschirm­foto 2022-10-25 um 11 19 45" src="https://user-images.githubusercontent.com/25504739/197737041-b72cb082-48d1-490d-90e4-7cb503a5dff2.png">

**Desktop (please complete the following information):**
 - OS: MacOs Ventura
 - Browser chrome
 - Version [106.0.5249.119 ]

**Additional context**

**Comment by ekondur** [comment]
Hi @madcoda9000 
Your properties are Pascal-Case and data Camel-Case I think,
Can you please check your serialization settings?

**Comment by madcoda9000** [comment]
Hi, @ekondur ,

thank you for your reply.

hmmm, that is courious. Because the Type <AppUser> (from what i use the poperties) is inherited from IdentityUser (from identity Framework). So there is no chance for me change or rename the properties.

The json data passed back is generated by ToDataResult. So i don't know what i could change here...

**Comment by ekondur** [comment]
Hi @madcoda9000 ,
It is not necessary to use Pascal-Case but it is an obligation that JSON and entity properties must be the same. The identity uses Pascal-Case I think.
Look at this please and don't hesitate to return back:
https://stackoverflow.com/questions/58392039/how-to-set-json-serializer-settings-in-asp-net-core-3

**Comment by madcoda9000** [comment]
Hello @ekondur ,

many thanks for your help. I've set the options in program.cs like that now:

```
builder.Services.AddControllersWithViews().AddJsonOptions(options =>
{
    options.JsonSerializerOptions.PropertyNamingPolicy = null;
});
```

And now the returned json property names are the same as the identity properties and the data is displayed in the datatable.

I just wanted to mention, that if i use vanilla "datatable js" it is only working without setting the serializer options. That means for the vanialla datatables js it is neccessary that all properties starts lowercase (which is documented in their project). So it seems that it is a standard that the json serializer returns properties starting lowercase. But with vanilla datatables js i am not able to use the html helpers, which is sad because i've written a lot :-)

Anyway. it is working now and I really appreciate your help.

Wish you All the best
Sascha

**Comment by madcoda9000** [comment]
ohh, just one question left (for me!). I try to archive something like that:
If data == null than display this else display data

I've tried like this:

`cols.Field(a=>a.ProfilePicture).Template("(data===null) ? '<div class=\"avatar avatar-md\"><img src=\"/assets/images/faces/user.png\" height=\"32\"></div>' : '<div class=\"avatar avatar-md\"><img src=\"' + data + '\" height=\"32\"></div>'");`

But then it displays only something when data is not null. If data is null it displays nothing?

<img width="1263" alt="Bildschirm­foto 2022-10-26 um 17 10 31" src="https://user-images.githubusercontent.com/25504739/198064679-b186dddd-d63e-4d65-a911-71dfa4f65749.png">

What did i made wrong?

Best Regards
Sascha

**Comment by ekondur** [comment]
Hi, @madcoda9000 
Thank you for your kind comments and useful suggestion.
Can you please try == instead of ===
Regards.

**Comment by madcoda9000** [comment]
Hello @ekondur ,

thanks for your reply. That did not work.

This:

<img width="1928" alt="Bildschirm­foto 2022-10-27 um 06 52 23" src="https://user-images.githubusercontent.com/25504739/198195043-8a691fcd-4351-4147-84e9-88164e220dda.png">



And this is the output:
<img width="1844" alt="Bildschirm­foto 2022-10-27 um 06 51 30" src="https://user-images.githubusercontent.com/25504739/198195059-b813207e-c8ad-4487-8720-f5452cb24b99.png">

I would expect that the output is like:

`'render': function(data, type, row, meta) { if (data == null) { return '<div class="avatar avatar-md"><img src="/assets/images/faces/user.png" height="32"></div>'; } else { return '<div class="avatar avatar-md"><img src="' + data + '" height="32"></div>'} ;`

I guess it is because of this (in DatatableJs/Helpers/RenderHelper.cs)

<img width="1312" alt="Bildschirm­foto 2022-10-27 um 07 31 13" src="https://user-images.githubusercontent.com/25504739/198199946-a7f423fc-79b9-434e-a656-ff10bd39cc39.png">


Best Regards
Sascha

**Comment by madcoda9000** [comment]
Hi @ekondur ,

me again (sorry for that). In the meantime i did a little research by myself.

I did the following:
1. remove the nuget packages to  DataTableJs and DataTableJs.Data from my Project
2. cloned your Repository locally 
3. made the following changes to JsHelper.cs

<img width="1562" alt="Bildschirm­foto 2022-10-27 um 10 07 54" src="https://user-images.githubusercontent.com/25504739/198228600-22151e0e-6a57-463c-bf11-f5ed328f2401.png">

4. build the DataTable projects
5. added Assembly References to my locally builded Datatable dll's


aaaand surpriiiiise (it's working now)

<img width="1901" alt="Bildschirm_foto 2022-10-27 um 10 13 32" src="https://user-images.githubusercontent.com/25504739/198264574-649baf6f-c5e3-475e-96f6-9bea65f302be.png">


Now using this code:

```
@(Html.JS().Datatable<AppUser>()
            .Name("PersonGrid")
            .Selecting(true)
            .Class("table table-striped")
            .Columns(cols =>
            {
                cols.Field(a => a.Id).Visible(false);
                cols.Field(a => a.ProfilePicture).Title("Avatar").Template("(data==null) ? '<div class=\"avatar avatar-md\"><img src=\"/assets/images/faces/user.png\" height=\"32\"></div>' : '<div class=\"avatar avatar-md\"><img src=\"' + data + '\" height=\"32\"></div>'");
                cols.Field(a=>a.UserName);
                cols.Field(a=>a.Email);
                cols.Field(a=>a.Id).Title("Roles").Template("'roles here'");
                cols.Field(a => a.Id).Title("2fa enabled").Template("(data==true) ? '<div class=\"form-check form-check-success\"><input class=\"form-check-input\" type=\"radio\"></div>' : '<div class=\"form-check form-check-danger\"><input class=\"form-check-input\" type=\"radio\"></div>'");
                cols.Field(a => a.IsMfaForce).Title("2fa enforced").Template("(data==false) ? '<div class=\"form-check form-switch\"><input class=\"form-check-input\" type=\"checkbox\" id=\"2fa' + row.Id + '\" data-uid=\"' + row.Id + '\" onclick=toggleMfaForce(\"' + row.Id + '\"); /></div>' : '<div class=\"form-check form-switch\"><input class=\"form-check-input\" type=\"checkbox\" id=\"2fa' + row.Id + '\" data-uid=\"' + row.Id + '\" checked=\"checked\"  onclick=toggleMfaForce(\"' + row.Id + '\"); /></div>'");
                cols.Field(a => a.IsLdapLogin).Title("Ldap").Template("(data==false) ? '<div class=\"form-check form-switch\"><input class=\"form-check-input\" type=\"checkbox\" id=\"ldap' + row.Id + '\" data-uid=\"' + row.Id + '\" onclick=toggleLdapLogin(\"' + row.Id + '\"); /></div>' : '<div class=\"form-check form-switch\"><input class=\"form-check-input\" type=\"checkbox\" id=\"ldap' + row.Id + '\" data-uid=\"' + row.Id + '\" checked=\"checked\"  onclick=toggleLdapLogin(\"' + row.Id + '\"); /></div>'");
                cols.Field(a => a.IsEnabled).Title("Active").Template("(data==false) ? '<div class=\"form-check form-switch\"><input class=\"form-check-input\" type=\"checkbox\" id=\"active' + row.Id + '\" data-uid=\"' + row.Id + '\" onclick=toggleLdapLogin(\"' + row.Id + '\"); /></div>' : '<div class=\"form-check form-switch\"><input class=\"form-check-input\" type=\"checkbox\" id=\"active' + row.Id + '\" data-uid=\"' + row.Id + '\" checked=\"checked\"  onclick=toggleLdapLogin(\"' + row.Id + '\"); /></div>'");
                cols.Field(a => a.Id).Title("Ops").Template("'<div class=\"btn-group\" role=\"group\" aria-label=\"role actions\"><a href=\"/Admin/EditUser/' + row.Id + '\" class=\"btn btn-primary btn-sm\">Edit</a>&nbsp;<a href=\"#\" class=\"btn btn-danger btn-sm\" onclick=DeleteUser(\"' + row.Id + '\");>Delete</a></div>'");

            })
            .URL(Url.Action("GetUsers", "Admin"), "POST")
            .ServerSide(true)
            .Render()
        )
```

It is looking damn nice :-)

<img width="1248" alt="Bildschirm_foto 2022-10-27 um 12 37 13" src="https://user-images.githubusercontent.com/25504739/198264618-f9452c29-be83-4988-a949-d013673d7a1d.png">


Best Regards
Sascha

**Comment by ekondur** [comment]
Thank you so much @madcoda9000 ,
I will check the problem and release a new nuget version if we need.
Regards.

**Comment by madcoda9000** [comment]
nothing to thank. I have to thank for this piece of software. Keep up that work!

**Comment by ekondur** [comment]
Hi @madcoda9000 
You can use it from now on,
```csharp
.URL(Url.Action("GetDataResult"), "POST", camelCase: true)
```
Set the camelCase property to `true`.
It is released with version [3.4.1](https://github.com/ekondur/DatatableJS/releases/tag/v3.4.1)
